internal/controllers/machinedeployment/machinedeployment_status_test.go

```diff
@@ -0,0 +1,1185 @@
+			getMachinesSucceeded: true,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineDeploymentDeletingV1Beta2Condition,
+				Status:  metav1.ConditionTrue,
+				Reason:  clusterv1.MachineDeploymentDeletingDeletionTimestampSetV1Beta2Reason,
+				Message: "Deleting 1 MachineSets",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			var machines collections.Machines
+			if tt.machines != nil {
+				machines = collections.FromMachines(tt.machines...)
+			}
+			setDeletingCondition(ctx, tt.machineDeployment, tt.machineSets, machines, tt.getAndAdoptMachineSetsForDeploymentSucceeded, tt.getMachinesSucceeded)
+
+			condition := v1beta2conditions.Get(tt.machineDeployment, clusterv1.MachineDeploymentDeletingV1Beta2Condition)
+			g.Expect(condition).ToNot(BeNil())
+			g.Expect(*condition).To(v1beta2conditions.MatchCondition(tt.expectCondition, v1beta2conditions.IgnoreLastTransitionTime(true)))
+		})
+	}
+}
+
+type fakeMachineSetOption func(ms *clusterv1.MachineSet)
+
+func fakeMachineSet(name string, options ...fakeMachineSetOption) *clusterv1.MachineSet {
+	p := &clusterv1.MachineSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+	for _, opt := range options {
+		opt(p)
+	}
+	return p
+}
+
+func withStatusReplicas(n int32) fakeMachineSetOption {
+	return func(ms *clusterv1.MachineSet) {
+		ms.Status.Replicas = n
+	}
+}
+
+func withStatusV1beta2ReadyReplicas(n int32) fakeMachineSetOption {
+	return func(ms *clusterv1.MachineSet) {
+		if ms.Status.V1Beta2 == nil {
+			ms.Status.V1Beta2 = &clusterv1.MachineSetV1Beta2Status{}
+		}
+		ms.Status.V1Beta2.ReadyReplicas = ptr.To(n)
+	}
+}
+
+func withStatusV1beta2AvailableReplicas(n int32) fakeMachineSetOption {
+	return func(ms *clusterv1.MachineSet) {
+		if ms.Status.V1Beta2 == nil {
+			ms.Status.V1Beta2 = &clusterv1.MachineSetV1Beta2Status{}
+		}
+		ms.Status.V1Beta2.AvailableReplicas = ptr.To(n)
+	}
+}
+
+func withStatusV1beta2UpToDateReplicas(n int32) fakeMachineSetOption {
+	return func(ms *clusterv1.MachineSet) {
+		if ms.Status.V1Beta2 == nil {
+			ms.Status.V1Beta2 = &clusterv1.MachineSetV1Beta2Status{}
+		}
+		ms.Status.V1Beta2.UpToDateReplicas = ptr.To(n)
+	}
+}
+
+type fakeMachinesOption func(m *clusterv1.Machine)
+
+func fakeMachine(name string, options ...fakeMachinesOption) *clusterv1.Machine {
+	p := &clusterv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+	for _, opt := range options {
+		opt(p)
+	}
+	return p
+}
+
+func withStaleDeletion() fakeMachinesOption {
+	return func(m *clusterv1.Machine) {
+		m.DeletionTimestamp = ptr.To(metav1.Time{Time: time.Now().Add(-1 * time.Hour)})
+	}
+}
+
+func withV1Beta2Condition(c metav1.Condition) fakeMachinesOption {
+	return func(m *clusterv1.Machine) {
+		if m.Status.V1Beta2 == nil {
+			m.Status.V1Beta2 = &clusterv1.MachineV1Beta2Status{}
+		}
+		meta.SetStatusCondition(&m.Status.V1Beta2.Conditions, c)
```

Comment:
nit: Maybe we should just append the condition vs using this func that sets the timestamp in a way that differs from our condition utils

Reply:
Used set